policies/coding-style.md

```diff
@@ -234,27 +234,44 @@ changing their context lines.
 
 ## Chapter 4: Naming
 
-C is a Spartan language, and so should your naming be. Do not use long
-names like _ThisVariableIsATemporaryCounter_. Use a name like _tmp_, which
-is much easier to write, and not more difficult to understand.
+C is a Spartan language, and so should your naming be.
 
-Except when otherwise required, avoid mixed-case names.
+Local variable names should be short, and to the point. If you have
+some random integer loop counter, it should probably be called `i` or `j`.
+Apart from that, avoid single-letter names,
```

Comment:
This is too strict! There might be cases where even a single letter is descriptive enough. Especially if the function is very short. I suggest: `Avoid single-letter names when they can be visually confusing, such as I and O.`

Reply:
I was told earlier that also names like `e`, `n`, and the like should be avoided,
i.e., that single-letter variable names should be avoided except for loop counters.
What do others think?

Reply:
Apart from being rather ambiguous and not descriptive, 
one-letter names have the drawback that searching for their occurrences (using a normal editor or browser) is a pain.

Reply:
local variable names like `e` and `n` are in my opinon very much fine if they are for example about the exponent e and modulus n in RSA.

> I was told earlier that also names like `e`, `n`, and the like should be avoided,

When that was?

Reply:
There are definitely instances in the code where it is confusing, e.g.

https://github.com/openssl/openssl/blob/802cacf34f2db9111becb4f0d3aa00460df13a19/ssl/t1_enc.c#L292-L314

Reply:
>> Apart from being rather ambiguous and not descriptive, 
one-letter names have the drawback that searching for their occurrences (using a normal editor or browser) is a pain.

> local variable names like `e` and `n` are in my opinion very much fine if they are for example about the exponent e and modulus n in RSA.

I agree that in such cases the names are telling.
Still be problem that it's hard to search occurrences stays.

> > I was told earlier that also names like `e`, `n`, and the like should be avoided,
> 
> When that was?

A couple of years back, when I contributed the CMP implementation.
I could try and did those reviewer comments out if this is helpful.

Reply:
> Still be problem that it's hard to search for occurrences stays.

But if it is a local variable, then you should not need to search for it in the whole source code context.

Reply:
> There are definitely instances in the code where it is confusing, e.g.
> 
> https://github.com/openssl/openssl/blob/802cacf34f2db9111becb4f0d3aa00460df13a19/ssl/t1_enc.c#L292-L314

That is NOT a short simple function, is it? And those single letter variable instances and even the two-letter ones are clearly not unambiguous.

Reply:
Yes, but even within a (not pretty short) function searching for names like `e` and `x` can be quite a pain.

Reply:
In that function I'd still probably keep using `s` as the `SSL *` parameter. Otherwise I am not sure I'd keep any other variable as single letter one an probably also rename a few others.

Reply:
So I would be OK with `Apart from that, avoid single-letter names if they are ambiguous or when they can be visually confusing...`

Reply:
How about this:
```
Avoid single-letter names when they can be visually confusing, such as `I` and `O`.
Avoid other single-letter names unless they are telling in the given context.
For instance, `m` for modulus and `s` for `SSL` pointers are fine.
```

Reply:
Good.

Reply:
Done.